DOL.WHD.Section14c.PdfAPi/PdfHelper/PdfHelper.cs

```diff
@@ -0,0 +1,274 @@
+        public static PdfDocument ConcatenatePDFs(PdfDocument outputDocument, ApplicationData applicationData)
+        {
+            if (!string.IsNullOrEmpty(applicationData.Type))
+            {
+                // Create PDF file
+                if (applicationData.Type.ToLower().Contains("pdf"))
+                {
+                    outputDocument = ConcatenatePDFDocumenByArray(outputDocument, applicationData.Buffer);
+                }
+
+                // Create PDF image
+                if (applicationData.Type.ToLower().Contains("image"))
+                {
+                    var doc = ConcatenatePDFWithImage(outputDocument, applicationData.Buffer);
+                    AddPagesToPdf(ref outputDocument, doc);
+                }
+
+                if (applicationData.Type.ToLower().Contains("html"))
+                {
+                    var doc = GetPdfDocFromHtml(outputDocument, applicationData.HtmlString);
+                    AddPagesToPdf(ref outputDocument, doc);
+                }
+            }
+
+            if (applicationData?.FilePaths != null)
+            {
+                outputDocument = ConcatenatePDFDocumentByPath(outputDocument, applicationData.FilePaths);
+            }
+
+            return outputDocument;
+        }
+
+        /// <summary>
+        /// Create PDF from HTML string
+        /// </summary>
+        /// <param name="outputDocument"></param>
+        /// <param name="htmlString"></param>
+        /// <returns></returns>
+        private static PdfDocument GetPdfDocFromHtml(PdfDocument outputDocument, string htmlString)
+        {
+            if (!string.IsNullOrEmpty(htmlString))
+            {
+                htmlString = ScrubHtmlString(htmlString);
+
+                PdfGenerateConfig config = new PdfGenerateConfig();
+                config.PageOrientation = PdfSharp.PageOrientation.Portrait;
+                config.PageSize = PdfSharp.PageSize.Letter;
+                config.SetMargins(30);
+
+                outputDocument = PdfGenerator.GeneratePdf(htmlString, PageSize.A4);
+            }
+
+            return outputDocument;
+        }
+
+        /// <summary>
+        /// Clean Html string and fix any html errors
+        /// </summary>
+        /// <param name="htmlString"></param>
+        /// <returns></returns>
+        private static string ScrubHtmlString(string htmlString)
+        {
+            string tempString = htmlString;
+            if (!string.IsNullOrEmpty(htmlString))
+            {
+                HtmlDocument doc = new HtmlDocument();
+                doc.LoadHtml(htmlString);
+                doc.OptionFixNestedTags = true;
+
+                List<HtmlNode> tdNodes = doc.DocumentNode.Descendants().Where(x => x.Name == "a" && x.Attributes.Contains("href") && x.Attributes["href"].Value.StartsWith("#")).ToList();
+               
+                foreach (HtmlNode node in tdNodes)
+                {
+                    node.Remove();
```

Comment:
How about I replace href="#" with href=""?

Reply:
That seems like it would work! I'm on my phone and don't have documentation in front of me, would you be able to delete the `href` attribute? Anyway, what you said is way easier than what I was thinking.